### PR TITLE
doc: add definition and process of WG

### LINKED ISSUE #33: Clarify the process of setting up a new working group under ORAS
We need a clarification of the process for creation of a working group under ORAS. The [GOVERNANCE.md](https://github.com/oras-project/community/blob/main/governance/GOVERNANCE.md#subproject-owners) specifies how maintainers are added but it doesn't specify how subprojects (WGs/repos) are created. We need to have the following answered:

- How a proposal for WG should be submitted?
- How is the proposal accepted? Who needs to approve the proposal to be accepted? Project maintainers? How many votes? Or just community? How many votes?
- What is the criteria for approval of a WG? Does this fit into the project chapter?
- What is required for the WG? New repo?
- How are maintainers for the  WG (and new repo) selected?
- What happens if maintainers in the WG do not participate in the WG? How maintainers are removed from the WG? 

There may be others but let's have the discussion in this issue.

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Fix #33 due to the WG proposal received a super-majority approval from the existing ORAS organization owners.

### CODE REVIEW

**Comment by sajayantony** [comment]
Testing if tagging @oras-project/oras-org-maintainers works. 
/cc @lachie83 @toddysm 

Thanks @FeynmanZhou for working on this. I'm hoping we can close on this soon. 
I couldn't interpret if there is a clause that enables the WG to recommend that as per its investigation, the recommendattion would be pursue working with the larger group outside of ORAS. 

For e.g. if the search working group decides to pursue working on CNCF tag post initial investigation/prototying etc. 

Reading these clauses means that the WG limited to 


> - an update to an existing specification (for example, adding support for a new set of features);
> - a new specification (for example, creating an specification for a new topical area); or
> - a new ORAS sub-project other than a specification (for example, creating a reference implementation of an specification).

File: governance/GOVERNANCE.md
Comment (sajayantony) [inline governance/GOVERNANCE.md:83]:
```suggestion
To create a Working Group in the ORAS community, organizers should have a clear goal measured through a specific deliverable or deliverables. Working Group is temporary in nature and will be eventually disbanded after reaching its stated goal(s).
```

Comment (sajayantony) [reply governance/GOVERNANCE.md:83]: Yes +1 - WG can produce code or a reference implementation that could be deliverable.

Comment (toddysm) [reply governance/GOVERNANCE.md:83]: @SteveLasker, I think we should say:
> the working group output _may_ not be a sustained code project? Any specs, code, recommendations would be donated to existing projects or new projects/sub-projects

If I am to take the existing proposal for the search WG, the reference implementation may live in the distribution sub-project or in zot. The API spec may or may not live in ORAS. _Will_ in your statement is too strong but changing to _may_ gives the necessary flexibility.

Hope this helps.

Comment (toddysm) [reply governance/GOVERNANCE.md:83]: Sorry, need to do too many edits. Here is what I think the wording should be:
> The working group output may not be a sustained sub-project. Any specs, code, and recommendations may be implemented in existing or new sub-projects or projects outside the scope of ORAS.

Comment (SteveLasker) [reply governance/GOVERNANCE.md:83]: For specs, I'm assuming that would be a sub-project. 
I'm also assuming we would fork Zot|Distribution into the oras-project, make the incubation changes there. If they play out as viable, they become an upstream PR. This is how we did the aritfacts-spec, when we made prototypes for implementation to distribution.
I really think the only difference between a working group and a sub-project is the expectation of long term release/support. This is why I'm not clear why a working group is anything more than a sub-project with expectations set for scope.

Comment (SteveLasker) [reply governance/GOVERNANCE.md:83]: The major differentation Sajay makes above is:

> temporary in nature and will be eventually disbanded after reaching its stated goal(s).

Is that effectively the only difference between a sub-project and working group? 

For instance:

> To create a Working Group in the ORAS community, organizers should have a clear goal measured through a specific deliverable or deliverables.

Wouldn't that also apply to sub-projects:

> To create a Subproject in the ORAS community, maintainers should have a clear goal measured through a specific deliverable or deliverables.

I might suggest a working group may not have clear goals. That's kinda the point. It's a place of incubation, to explore some ideas, like a startup that starts with one idea with a sore set of like-minded people. They may pivot a few times, and that's ok, and actually, the freedom to land a great user-focused solution. 

Even sub-projects pivot as they learn and explore new scenarios. I think we should be careful to not overly limit the creativity. We might want to put some scope barriers up, only to set expectations to come back to the "org owners" for approval to expand charter.

Comment (toddysm) [reply governance/GOVERNANCE.md:83]: @SteveLasker you are looking at the very narrow work that is currently considered for search. The definition of the WG will allow us to do work in the future that doesn't end up in any sub-project but may directly be donated to upstream. Also, I personally cannot say that what you are describing above **_will be_** the approach for search **_before_** we complete the work in the working group. Unless there have been already discussions on the design that I am not aware of. 

For example, the working group may decide that Zot will implement the search directly in their repo and there is no need to fork into ORAS (cc:// @rchincha). The working group may decide that the spec should not be written under ORAS but is better to collaborate with another CNCF community. So, in this case there is no need for sub-projects in ORAS at all. We should consider those possibilities for the search as well as for any other proposal we decide to drive as part of ORAS in the future.

Comment (FeynmanZhou) [reply governance/GOVERNANCE.md:83]: Agree with @SteveLasker @sajayantony , saying `The Working Group does not own any code.` is not exact and we need to remove this description.

**Comment by SteveLasker** [comment]
I appreciate some super large projects like the kubernetes project may have working groups, but those are significantly larger. Other projects like Helm have [Helm Improvement Proposals (HIPS)](https://github.com/helm/community/blob/main/hips/README.md) for ideas that enable iteration.

Some more info related to CNCF Working groups, and how they are a component of a sub-project, but not yet another thing. 
- [CNCF Governance By Subprojects Template](https://contribute.cncf.io/maintainers/templates/governance-subprojects/)
  - >For example, you can pretty easily decide that each “working group” in your project is a subproject
  - >**Subproject**: The individual group/repository where work on the project gets done; alternately called projects, SIGs, repos, drivers, plugins, operators, working groups, or other units of work that each have their own maintainers.

However, CNCF does have working groups, that aren't tied to a particular project:
- [Contribution Guides to the CNCF Ecosystem - Working Groups](https://contribute.cncf.io/contributors/#working-groups)

The current search effort is focusing on the CNCF TAG to iterate. As a result, I'm not seeing good justification to complicate the ORAS governance with a working-group definition, above what a sub-project is already doing.

**Comment by FeynmanZhou** [comment]
@SteveLasker @sajayantony @toddysm Thanks for providing your suggestions. I see the biggest confusion seems to be lacking clarity about the difference between Sub-project and the Working Group. I would like to elaborate on the major difference and justifications for both after I investigated a few more CNCF projects:

## Differences between Sub-project and the Working Group

Working groups are organizations responsible for the design and implementation of large architectural aspects of the overall ORAS project. Working groups operate with a fair amount of autonomy within the broader scope of the project. They tend to be long-lived or temporary. 

The sub-project is more focused on implementation and solving specific requirements. It could be the outcome of the Working Group. A Working Group's scope may be across multiple sub-projects.

## Working Groups are running in multiple CNCF projects.

Apart from the latest Kubernetes community in CNCF, WGs are also popular and are actively running in other CNCF projects/communities either. 

- [Knative](https://github.com/knative/community/blob/main/working-groups/WORKING-GROUPS.md)
- [Istio](https://github.com/istio/community/blob/master/WORKING-GROUPS.md)
- [KubeEdge](https://github.com/kubeedge/community/tree/master/wg-wireless)

**Comment by SteveLasker** [comment]
Thanks @FeynmanZhou,
There's a lot of great detail for working groups. Much, if not most, or possibly all would seem to apply to sub-projects. 
Before we complete the delta definition of a working group, can more clarity be created around a sub-project?
That might help understand what/why a new governance would be required.

**Review by sajayantony — APPROVED** [review]
Approving this as a baseline and recommend we add/modify as needed.

**Review by TerryHowe — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by FeynmanZhou** [comment]
@sabre1041 Looking forward to your opinion on this WG process proposal.

**Review by sabre1041 — COMMENTED** [review]
This looks really good. However, one area that I would request additional clarification/discussion is surrounding the length of time for a working group.

In the provided text, it states that a WG can either be long lived or temporary. Working groups should be short lived in order to provide a concrete scope of work that should be accomplished. I feel like the text should read short lived rather than long lived.

The rest of the content looks great.

**Review by sajayantony — APPROVED** [review]

**Comment by FeynmanZhou** [comment]
@sabre1041 It makes sense to me. I revised the WG description at [line 65](https://github.com/oras-project/community/pull/48/files#diff-4b231353191f8b30dc252aede7ce6a4fb33cdb5c71c75634a905719fd79ac462R60) and clarified WG tends to be short-lived.

Sorry for the late response. I just back from a holiday yesterday. We can continue to work on ORAS things.

**Review by TerryHowe — APPROVED** [review]

**Review by sabre1041 — APPROVED** [review]
LGTM

**Comment by sajayantony** [comment]
Added to agenda for tomorrow. /cc @SteveLasker

**Comment by SteveLasker** [comment]
I’m on vacation, and defer to the majority of the maintainers for next steps

**Pull request merged**
